gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobScheduler.java

```diff
@@ -431,7 +431,7 @@ private void cancelJobIfRequired(DeleteJobConfigArrivalEvent deleteJobArrival) t
       if (jobNameToWorkflowIdMap.containsKey(deleteJobArrival.getJobName())) {
         String workflowId = jobNameToWorkflowIdMap.get(deleteJobArrival.getJobName());
         TaskDriver taskDriver = new TaskDriver(this.jobHelixManager);
-        taskDriver.waitToStop(workflowId, this.helixJobStopTimeoutMillis);
+        taskDriver.stop(workflowId);
```

Comment:
What are the consequences of stop failing (not timing out) and we start a new job without verifying if the previous job has been successfully stopped or not?

Reply:
That's the bug I'm trying to fix. Currently, when handleUpdateJobConfigArrival, firstly we try to delete the old job. The waitToStop always timeout, and I can't find helix SLA for it. This results the old job actually got killed, but in our jobRunningMap it never get updated. Thus, the new job won't be launched as the job launcher thinks there's still a running one.

Reply:
My change make the helix workflow stop async, and will check the job status by waitForJobCompletion, so can avoid the bug I mentioned above.

Reply:
If the stop failed, we still launch a new one, the later one will not run since we check the jobRunningMap if there's any existing job.

Reply:
So should we update the `jobRunningMap` if `stop` timeouts? 
Also, we should check with helix why it is timing out, are we giving too less of a timeout? If the expected time to stop is really too much, like 1+ min we can switch to asynchronous stop.

Reply:
I think the issue actually resides in our codebase, but won't be a strait-forward fix. The jobRunningMap is parsed to GobblinHelixJobLauncher, which will constantly pull the helix workflow status and will update the map based on it. I find current behavior is whenever we call waitToStop, the workflow goes to stopping state, and [HelixUtils.waitJobCompletion](https://github.com/apache/gobblin/blob/8c9c8a84ed23c0215c4d80125ac532e97085d76f/gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixUtils.java#L278) will then delete it, so it never really goes to stopped state.

Reply:
So my change relies on GobblinHelixJobLauncher to update the jobRunningMap when it realize the job got terminated,  and in GobblinHelixJobScheduler.waitForJobCompletion() we check the map after calling stop the workflow.

Reply:
In that case, why not make `GobblinHelixJobLauncher` update the `jobRunningMap` when Helix API timeouts?

Reply:
The issue is that waitToStop is called by GobblinHelixJobScheduler, and GobblinHelixJobLauncher as a separate thread doesn't know that Helix API call status.

Reply:
But just above you confirmed that _"The jobRunningMap is parsed to GobblinHelixJobLauncher"_
They both have access to this map, which can be updated incase of helix api timeouts.

Reply:
1. the api timeout doesn't mean delete is successful 
2. Though GobblinHelixJobLauncher has the reference of jobRunningMap, it doesn't know the result of the api call. It only update the map when job [finished](https://github.com/apache/gobblin/blob/2463eda692b97d6947b21154abd37997a9ac3a73/gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java#L458)